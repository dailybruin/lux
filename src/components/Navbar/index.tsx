import { faBars, faTimes } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import * as React from 'react'
import { css } from 'react-emotion'
import logoSrc from './db-logo.png'

interface NavbarProps {
  /** Array of React elements. */
  links?: React.ReactNode[]
  /** custom style applied to individual links. */
  linkStyle?: string
  /** Logo can be text or an image. */
  showLogo: boolean
  /** Site title. */
  title: string
  /** Logo link. */
  logoURL: string
  /** Pins navbar (or not). */
  sticky?: boolean
  /** custom style applied to the navbar. */
  style?: string
}

class Navbar extends React.Component<NavbarProps> {
  public static defaultProps = {
    logoURL: '/',
    sticky: false,
  }

  public state = {
    open: false,
  }

  public handleClick = () => {
    this.setState({ open: !this.state.open })
  }

  public render() {
    const sticky = css`
      position: sticky;
      top: 0;
    `

    const navbarStyle = css`
      box-shadow: 0 2px 2px -2px rgba(0, 0, 0, 0.15);
      padding: 10px;
      display: flex;
      justify-content: space-between;
      width: auto;
      background-color: white;
      ${this.props.sticky ? sticky : ''};
      ${this.props.style};
    `
    const linkStyle = css`
      @media (min-width: 940px) {
        padding-right: 3rem;
        align-self: center;
      }
      ${this.props.linkStyle};
    `
    const linksMobileStyle = css`
      display: flex;
      flex-direction: column;
      justify-content: flex-end;
      align-items: flex-end;
    `

    const logoStyle = css`
      a {
        color: inherit; /* blue colors for links too */
        text-decoration: inherit; /* no underline */
        font-family: inherit;
        font-weight: inherit;
        font-size: inherit;
      }
      img {
        overflow: hidden;
        vertical-align: bottom;
        margin-right: 20px;
        width: 50px;
      }
      display: flex;
      align-items: center;
    `
    const linksStyle = css`
      display: flex;
      justify-content: space-between;
      @media screen and (max-width: 940px) {
        display: none;
      }
    `

    const logo = (
      <a href={this.props.logoURL}>
        {this.props.showLogo && <img src={logoSrc} />}
      </a>
    )

    const links = this.props.links
      ? this.props.links.map((link, i) => {
          return (
            <div key={i} className={linkStyle}>
              {link}
            </div>
          )
        })
      : undefined

    return (
      <div className={navbarStyle}>
        <div className={logoStyle}>
          {this.props.showLogo && logo}
          <h2>{this.props.title}</h2>
        </div>
        <div className={linksStyle}>{links}</div>
        <details
          className={css`
            width: 100%;
            justify-self: end;
            text-align: right;
            font-size: 2rem;
            @media (min-width: 940px) {
              display: none;
            }
          `}
        >
          <summary
            className={css`
              font-size: 1rem;
              ::-webkit-details-marker {
                display: none;
              }
            `}
          >
            <div onClick={this.handleClick}>
              {' '}
              <FontAwesomeIcon icon={this.state.open ? faTimes : faBars} />{' '}
            </div>
          </summary>
          <div className={linksMobileStyle}>{links}</div>
        </details>
      </div>
    )
  }
}

export default Navbar
