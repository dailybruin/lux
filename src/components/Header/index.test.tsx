import * as React from 'react'
import { cleanup, render } from 'react-testing-library'
import Navbar from '.'

describe('Navbar', () => {
  afterEach(cleanup)

  it('Matches its snapshot', () => {
    const { container } = render(
      <Navbar
        sticky={true}
        showLogo={true}
        title={'Fake Title'}
        links={[
          <a key={1} href="/">
            Index
          </a>,
          <a key={2} href="/page-two">
            Page Two
          </a>,
          <a key={3} href="/page-three">
            Page Three
          </a>,
        ]}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
