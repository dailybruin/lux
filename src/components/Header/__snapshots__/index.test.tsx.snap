// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Matches its snapshot 1`] = `
.emotion-6 {
  display: grid;
  grid-template-columns: 1fr 1fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 1.1rem 1.2rem;
  background-color: white;
  box-shadow: 0 2px 2px -2px rgba(0,0,0,0.15);
  position: -webkit-sticky;
  position: sticky;
  top: 0;
}

.emotion-1 {
  justify-self: start;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.emotion-0 {
  font-weight: bold;
  text-transform: uppercase;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (max-width:940px) {
  .emotion-2 {
    display: none;
  }
}

.emotion-5 {
  width: 100%;
  justify-self: end;
  text-align: right;
  font-size: 2rem;
}

@media (min-width:940px) {
  .emotion-5 {
    display: none;
  }
}

.emotion-3 {
  font-size: 1rem;
}

.emotion-3::-webkit-details-marker {
  display: none;
}

.emotion-4 {
  display: grid;
  grid-template-columns: 1fr;
}

<header
  class="emotion-6"
>
  <a
    class="emotion-1"
    href="/"
  >
    <span
      class="emotion-0"
    >
      Fake Title
    </span>
  </a>
  <nav
    class="emotion-2"
  >
    <a
      href="/"
    >
      Index
    </a>
    <a
      href="/page-two"
    >
      Page Two
    </a>
    <a
      href="/page-three"
    >
      Page Three
    </a>
  </nav>
  <details
    class="emotion-5"
  >
    <summary
      class="emotion-3"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bars fa-w-14 "
        data-icon="bars"
        data-prefix="fas"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          fill="currentColor"
        />
      </svg>
    </summary>
    <nav
      class="emotion-4"
    >
      <a
        href="/"
      >
        Index
      </a>
      <a
        href="/page-two"
      >
        Page Two
      </a>
      <a
        href="/page-three"
      >
        Page Three
      </a>
    </nav>
  </details>
</header>
`;
